pegasus/sites.v3/hourofcode.com/public/events/2015/index.haml

```diff
@@ -0,0 +1,40 @@
+---
+title: <%= hoc_s(:events_all_title) %>
+layout: wide
+nav: events_nav
+---
+-SOLR = Solr::Server.new(host:CDO.solr_server)
+
+-by_country = {}
+-by_state = {}
+-country_names = {}
+-state_names = {}
+
+-events = SOLR.query(q:"*:*", fq:["kind_s:HocSignup2015", "location_country_s:[* TO *]", "location_country_code_s:[* TO *]", "organization_name_s:[* TO *]"], sort:'location_country_s asc, location_state_code_s asc', rows:1000)
```

Comment:
This is just a philosophical observation about much of Pegasus.. Rails highly encourages the use of controllers and models to hoist code out of views.. that way you can test the view more easily with dummy data, and even more valuable, test your logic without a view being rendered through unit-tests.  No need to make drastic changes now, but I definitely appreciate the value of these aspects of MVC.